CyclopsNuclearReactor/CyNukeReactorMono.cs

```diff
@@ -0,0 +1,383 @@
+
+        public override void Awake()
+        {
+            base.Awake();
+
+            if (_buildable == null)
+            {
+                _buildable = GetComponentInParent<Constructable>();
+            }
+
+            if (_saveData == null)
+            {
+                string id = GetComponentInParent<PrefabIdentifier>().ClassId; // Changed this to classId because Id was returning null
+                _saveData = new CyNukeReactorSaveData(id, MaxSlots);
+            }
+
+            if (RodsContainer == null)
+            {
+                InitializeRodsContainer();
+            }
+        }
+
+        private void Start()
+        {
+            SubRoot cyclops = GetComponentInParent<SubRoot>();
+
+            if (cyclops is null)
+            {
+                QuickLogger.Debug("Could not find Cyclops during Start. Attempting external syncronize.");
+                CyNukeChargeManager.SyncReactors();
+            }
+            else
+            {
+                QuickLogger.Debug("Parent cyclops found directly!");
+                ConnectToCyclops(cyclops);
+            }
+        }
+
+        public void ConnectToCyclops(SubRoot cyclops, CyNukeChargeManager manager = null)
+        {
+            ParentCyclops = cyclops;
+            this.transform.SetParent(cyclops.transform);
+
+            Manager = manager ?? CyNukeChargeManager.GetManager(cyclops);
+
+            if (!Manager.CyNukeReactors.Contains(this))
+                Manager.CyNukeReactors.Add(this);
+
+            QuickLogger.Debug("Cyclops Nuclear Reactor has been connected", true);
+        }
+
+        private void InitializeRodsContainer()
+        {
+            QuickLogger.Debug("Initializing Storage");
+            if (_rodsRoot == null)
+            {
+                var storageRoot = new GameObject("StorageRoot");
+                storageRoot.transform.SetParent(this.transform, false);
+                _rodsRoot = storageRoot.AddComponent<ChildObjectIdentifier>();
+            }
+
+            RodsContainer = new ItemsContainer(2, 2, _rodsRoot.transform, CyNukReactorSMLHelper.StorageLabel(), null);
+            RodsContainer.SetAllowedTechTypes(new[] { TechType.ReactorRod, TechType.DepletedReactorRod });
+
+            RodsContainer.isAllowedToAdd += (Pickupable pickupable, bool verbose) =>
+            {
+                TechType techType = pickupable.GetTechType();
+
+                return techType == TechType.ReactorRod || (isLoadingSaveData && techType == TechType.DepletedReactorRod);
+            };
+
+            RodsContainer.isAllowedToRemove += (Pickupable pickupable, bool verbose) =>
+            {
+                return pickupable.GetTechType() == TechType.DepletedReactorRod;
+            };
+
+            RodsContainer.onAddItem += OnAddItem;
+            RodsContainer.onRemoveItem += OnRemoveItem;
+        }
+
+        #endregion
+
+        #region Save Data
+
+        public void OnProtoDeserialize(ProtobufSerializer serializer)
```

Comment:
Compare to the [CyBioReactor](https://github.com/PrimeSonic/PrimeSonicSubnauticaMods/blob/master/MoreCyclopsUpgrades/Monobehaviors/CyBioReactorMono.cs) and see why save data isn't loading.

Reply:
Resolved. The issue was that the implementations of `OnProtoDeserialize` and `OnProtoDeserializeObjectTree` were backwards.